security/pkg/pki/ca/ca.go

```diff
@@ -57,6 +57,8 @@ const (
 type CertificateAuthority interface {
 	// Sign generates a certificate for a workload or CA, from the given CSR and TTL.
 	Sign(csrPEM []byte, ttl time.Duration) ([]byte, error)
+	// SignCAServerCert generates a certificate for the CA server (to serve the CSR).
```

Comment:
Should we extract this out of interface and we add `forCA bool` in Sign() method, letting the server call Sign() with different argument? Or leave a TODO for now?

Reply:
Will add a TODO here.